### ISSUE TITLE
Status for v0.3.0

### ISSUE THREAD

**Issue by akshathmangudi** [refrakt-hub/refrakt_core#2]
## Status for v0.3.0

Major changes will be introduced within `refrakt_core` for the third iteration. 
- [x] Functional models: 
    - [x] ResNet (18, 50, 101, 152)
    - [x] ConvNeXt
    - [x] Autoencoder (Simple, VAE) 
    - [x] Swin Transformer
    - [x] Vision Transformer
    - [x] SRGAN
    - [x] SimCLR
    - [x] DINO
    - [x] Masked Autoencoder
    - [x] Masked Siamese Networks
- [x] Features: 
    - [x] `sklearn`/`cuML` True Fusion integration. 
    - [x] Modularize larger files (codes larger than 200) and split functionalities to simplify train/test and inference scripts.
    - [x] Improve registries and logging configurations for safe import and avoiding global variables.  
    - [x] Smooth integration with `wandb` and `tensorboard`.
    - [x] Unify all training configurations inside yaml files to allow train/test/inference and pipeline modes by running `refrakt --config /path/to/config.yaml`
    - [x] run `radon cc` and `lizard` tests and refactor the code-base to not exceed CC threshold. 
    - [x] Ensure artifact dumping and logging to `tensorbord`/`wandb` works for all models (fusion, dl, ml pipelines)
    - [x] Allow manual overrides to hyperparameters for better flexibility. 
    - [x] Include a dynamic dataset loader that can take any custom.zip file and be able to either parse it or debug the incorrect format. 
    - [x] Add standard image resizing capabilities to prevent difficulty in loading images. If a non-standard image is provided, throw an exception for the user to resize the image if it is too large.
- [x] Software Development: 
    - [x] Create a `pre-commit.yaml` file for linting, formatting and type checking errors. 
    - [x] Include extensive sanity checks, smoke, unit and integration tests to improve robustness. 
    - [x] Setup `doctype` for refrakt_core.

**Comment by akshathmangudi** [comment]
current training status as of June 19th, 2025: 
- unified training configuration to run `refrakt --config ./path/to/config.yaml`
- smooth integration with `wandb` and `tensorboard` supported with interactive gradient distribution plots and model graphs. 
- supported models that use ModelOutput and LossOutput are: simclr, dino, resnet, convnext, autoencoder variants and mae. 

[NOTE] dynamic resizing has been deferred until further notice.

**Comment by akshathmangudi** [comment]
current status as of July 3rd, 2025: 
- `sklearn`/`cuml` integration complete. current ml head supported are: `RandomForestClassifier`, `LogisticRegression`, `SVC`, `KNeighborsClassifier`, `XGBClassifier`
- Larger files like `train.py`, `test.py` and `inference.py` have been modularized to contain less than 200 LOC, and helper functions have been created inside `api/utils/` for the above files. 

check latest commit under `feature/docs_checking` branch

**Comment by akshathmangudi** [comment]
current status as of July 5th, 2025: 
- switching to `feature/resizer_and_handler` to take care of image resizer and dataset handler capabilities. 
- codebase has been mostly type-enforced with strict mode enabled.

**Comment by akshathmangudi** [comment]
current status as of July 9th, 2025: 
- switched to `feature/pre_v3_testing` for final checks and code health.
- the entire codebase (except for the tests) have been type-enforced using `mypy.ini` 
- currently passes: `isort`, `black` and `mypy` tests.
- wip: `pylint`, `ruff`and a few files that need to be doctyped. but, the majority of the work is almost complete. 

estimated time to staging: by july 16th.

**Comment by akshathmangudi** [comment]
current status of July 12th, 2025: 

**refrakt v0.3.1** is out on PyPI. closing issue.
